src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -22,21 +22,23 @@ import Daml.Finance.Interface.Lifecycle.Effect qualified as Effect (I)
 import Daml.Finance.Interface.Lifecycle.Event qualified as Event (I)
 import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (I)
 import Daml.Finance.Interface.Types.Common (Id, Parties)
-import Daml.Finance.Interface.Types.Date.Calendar (BusinessDayAdjustment(..))
+import Daml.Finance.Interface.Types.Date.Calendar
 import Daml.Finance.Interface.Types.Date.Classes (toUTCTime)
 import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.RollConvention
 import Daml.Finance.Interface.Types.Date.Schedule (Frequency(..), PeriodicSchedule(..), Schedule, SchedulePeriod)
 import Daml.Finance.Interface.Types.Fpml
 import Daml.Finance.Interface.Util.Disclosure qualified as Disclosure (I, flattenObservers)
 import Daml.Finance.Lifecycle.Effect (Effect(..))
 import Daml.Finance.Lifecycle.Types (Unit(..))
+import Daml.Finance.Util.Date.Calendar (addBusinessDays, merge)
 import Daml.Finance.Util.Date.DayCount (calcPeriodDcf)
 import Daml.Finance.Util.Date.Schedule (createSchedule)
 import Prelude hiding (key)
 
 -- | Retrieve holiday calendar(s) from the ledger and roll out the payment schedule.
-rollPaymentSchedule : PeriodicSchedule -> [Text] -> Party -> Party -> Update Schedule
-rollPaymentSchedule periodicSchedule holidayCalendarIds issuer calendarDataAgency = do
+getHolidayCalendars : [Text] -> Party -> Party -> Update [HolidayCalendarData]
```

Comment:
Wrong description?

Reply:
Thanks! Fixed in the new PR: https://github.com/digital-asset/daml-finance/pull/246